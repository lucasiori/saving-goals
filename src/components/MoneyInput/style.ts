import styled from 'styled-components';
import NumberInput from 'react-number-format';

const Wrapper = styled.div`
  position: relative;
  height: 100%;
  border: ${({ theme }) =>
    `${theme.borders.width.sm} solid ${theme.colors.blueGray['50']}`};
  border-radius: ${({ theme }) => theme.borders.radius.sm};

  img {
    position: absolute;
    top: 50%;
    left: ${({ theme }) => theme.spacings.xs};
    transform: translateY(-50%);
  }
`;

const Input = styled(NumberInput)`
  width: 100%;
  height: 100%;
  border: 0;
  font-family: ${({ theme }) => theme.font.families.heading};
  color: ${({ theme }) => theme.colors.blueGray['600']};
  font-size: ${({ theme }) => theme.font.sizes.md};
  font-weight: ${({ theme }) => theme.font.weights.medium};
  padding: ${({ theme }) => `0.875rem ${theme.spacings.xxs}`};
  padding-left: 2.75rem;
`;

export { Wrapper, Input };
