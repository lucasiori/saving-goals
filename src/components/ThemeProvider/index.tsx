import { ThemeProvider } from 'styled-components';
import theme from '../../styles/theme';

interface ThemeProviderProps {
  children: React.ReactNode;
}

export default function ThemeProviderCP({
  children,
}: ThemeProviderProps): JSX.Element {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
}
