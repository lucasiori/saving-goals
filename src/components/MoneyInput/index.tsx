import { NumberFormatValues } from 'react-number-format';
import dollarSignSvg from '../../assets/icons/dollar-sign.svg';
import { Wrapper, Input } from './style';

interface MoneyInputProps {
  id: string;
  onChange?: (value?: number) => void;
}

export default function MoneyInputCP({
  id,
  onChange,
}: MoneyInputProps): JSX.Element {
  function handleChangeValue(value: NumberFormatValues) {
    if (onChange) {
      onChange(value.floatValue);
    }
  }

  return (
    <Wrapper>
      <img src={dollarSignSvg} alt="Dollar" width="24px" />

      <Input
        data-id="money-input"
        type="text"
        id={id}
        name={id}
        thousandSeparator=","
        allowNegative={false}
        onValueChange={handleChangeValue}
      />
    </Wrapper>
  );
}
