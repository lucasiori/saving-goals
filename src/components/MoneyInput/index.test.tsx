import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import ThemeProvider from '../ThemeProvider';
import MoneyInput from '.';

describe('MoneyInput', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <ThemeProvider>
        <MoneyInput id="money-input-test" />
      </ThemeProvider>
    );
    const component = tree.toJSON();

    expect(component).toMatchSnapshot();
  });

  describe('when inputs a value', () => {
    it('should trigger onChange function', () => {
      const onChange = jest.fn();

      const component = mount(
        <ThemeProvider>
          <MoneyInput id="money-input-test" onChange={onChange} />
        </ThemeProvider>
      );

      component.find('input[data-id="money-input"]').simulate('focus');

      component
        .find('input[data-id="money-input"]')
        .simulate('change', { target: { value: '25000', focus: () => {} } });

      expect(onChange).toBeCalledWith(25000);
    });
  });
});
