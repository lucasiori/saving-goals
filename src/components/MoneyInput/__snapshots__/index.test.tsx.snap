// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoneyInput renders correctly 1`] = `
<div
  className="sc-bdnxRM dKKJJO"
>
  <img
    alt="Dollar"
    src="dollar-sign.svg"
    width="24px"
  />
  <input
    className="sc-gtsrHT wdgxn"
    data-id="money-input"
    id="money-input-test"
    inputMode="numeric"
    name="money-input-test"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onMouseUp={[Function]}
    type="text"
    value=""
  />
</div>
`;
